src/Symfony/Component/Validator/Constraints/Collection.php

```diff
@@ -41,9 +41,12 @@ class Collection extends Composite
      */
     public function __construct($fields = null, array $groups = null, $payload = null, bool $allowExtraFields = null, bool $allowMissingFields = null, string $extraFieldsMessage = null, string $missingFieldsMessage = null)
     {
-        // no known options set? $fields is the fields array
-        if (\is_array($fields)
-            && !array_intersect(array_keys($fields), ['groups', 'fields', 'allowExtraFields', 'allowMissingFields', 'extraFieldsMessage', 'missingFieldsMessage'])) {
```

Comment:
I'm pretty sure `payload` was missing in the array of known options. Switching to a dynamic approach using reflection should solve this and any similar issues that may occur in the future.

Reply:
What about picking the first item of `$fields` instead and check if it is a `Constraint` instance (or an array of)?

Reply:
Won't work cause of https://github.com/symfony/symfony/blob/ddf3bcbbb2ec1b4110b9998b2414dee2091def36/src/Symfony/Component/Validator/Constraints/Collection.php#L69-L74

Reply:
I am not sure I understand. Checking for instances of `Constraint` would still yield `true`, wouldn't it?

Reply:
Yes, I don't recall why I initially discarded the suggestion, but it works. I did have to modify one test because a different exception is thrown now. I'm not sure if that qualifies as a BC break.